src/modsim2/attack/adversarial_images.py

```diff
@@ -71,8 +73,21 @@ def generate_adversarial_images(
     if attack_fn_name not in ["L2FastGradientAttack", "BoundaryAttack"]:
         raise NotImplementedError("This attack has not been implemented")
 
+    # If needed, auto select pytorch device/accelerator
+    if device == "auto":
+        device = choose_auto_accelerator()
+    # Depenency chain leads to float64 being created which is incompatible with MPS
```

Comment:
could you elaborate in an issue that we link (even if we don't intend to solve it ourselves)

Reply:
I've raised #44 and linked to it in the comment